src/centengine/engine_impl.cc

```diff
@@ -95,6 +95,66 @@ grpc::Status engine_impl::GetStats(grpc::ServerContext* /*context*/,
   return grpc::Status::OK;
 }
 
+grpc::Status engine_impl::GetHost(grpc::ServerContext* context,
+                                  const HostIdentifier* request,
+                                  EngineHost* response) {
+  std::promise<std::pair<bool, EngineHost*>> hostpromise;
+  std::future<std::pair<bool, EngineHost*>> f1 = hostpromise.get_future();
+  EngineHost host;
+
+  auto lambda = [&hostpromise, request, &host]() -> int32_t {
+    std::shared_ptr<com::centreon::engine::host> selectedhost;
+    std::unordered_map<std::string,
+                       std::shared_ptr<com::centreon::engine::host>>::iterator
+        ithostname;
+    std::unordered_map<uint64_t,
+                       std::shared_ptr<com::centreon::engine::host>>::iterator
+        ithostid;
+
+    switch (request->identifier_case()) {
+      case HostIdentifier::kHostName:
+        ithostname = host::hosts.find(request->host_name());
+        if (ithostname != host::hosts.end()) {
+          selectedhost = ithostname->second;
+        } else {
+		  hostpromise.set_value(std::make_pair(false, nullptr));
+          return (1);
```

Comment:
"return" is not a function, so we try to write "return 1" and not "return (1)".
Be careful, the identation is bad line 120.